services/core-web/src/components/mine/Permit/PermitConditionLayer.tsx

```diff
@@ -1,32 +1,206 @@
+import { createMineReportPermitRequirement } from "@mds/common/redux/slices/mineReportPermitRequirementSlice";
 
 interface PermitConditionLayerProps {
   condition: IPermitCondition;
   level?: number;
   isExpanded?: boolean;
+  setParentExpand?: () => void;
 }
 
 const PermitConditionLayer: FC<PermitConditionLayerProps> = ({
   condition,
   isExpanded,
   level = 0,
+  setParentExpand = () => {},
 }) => {
-  const expandClass = isExpanded ? "condition-expanded" : "condition-collapsed";
+  const dispatch = useDispatch();
+  const { id: mineGuid, permitGuid } = useParams<{ id: string; permitGuid: string }>();
+
+  const [isEditMode, setIsEditMode] = useState<boolean>(false);
+  const [expandClass, setExpandClass] = useState(
+    isExpanded ? "condition-expanded" : "condition-collapsed"
+  );
   const className = `condition-layer condition-layer--${level} condition-${condition.condition_type_code} fade-in`;
+
+  const handleSetParentExpand = () => {
+    if ((level = 0)) {
+      return;
+    } else {
+      setExpandClass("condition-expanded");
+      setParentExpand();
+    }
+  };
+
+  useEffect(() => {
+    setExpandClass(isExpanded || isEditMode ? "condition-expanded" : "condition-collapsed");
+  }, [isExpanded]);
+
+  const handleSectionClick = (event) => {
+    event.stopPropagation();
+    setParentExpand();
+    setIsEditMode(true);
+  };
+
+  const closeEdit = (event) => {
+    event.stopPropagation();
+    setIsEditMode(false);
+  };
+
+  const addNewReport = async (values) => {
+    await dispatch(createMineReportPermitRequirement({ mineGuid, values }));
+    dispatch(fetchPermits(mineGuid));
+    dispatch(closeModal());
+  };
+
+  const handleOpenAddReportModal = (event, reportCondition: IPermitCondition) => {
+    event.stopPropagation();
+    dispatch(
+      openModal({
+        props: {
+          onSubmit: addNewReport,
+          title: `Add Permit Required Report to Condition`,
+          condition: reportCondition,
+          permitGuid,
+        },
+        content: ReportPermitRequirementForm,
+      })
+    );
+  };
+
+  const sectionEdit = isEditMode || false;
+
   return (
-    <div className={className}>
+    <div
+      className={`${className} ${isEditMode ? "condition-layer--editing" : ""}`}
+      onClick={handleSectionClick}
+      onKeyPress={handleSectionClick}
+    >
       <div className={expandClass}>
-        <p className={className}>
+        <p>
           {condition.step} {condition.condition}
         </p>
-        {condition?.sub_conditions?.map((condition) => {
+        {sectionEdit && (
+          <Row justify="space-between" align="middle">
+            <Col>
+              <Row gutter={8} className="condition-edit-buttons" align="middle">
+                <Col>
+                  <Button
+                    onClick={(event) => {
+                      event.stopPropagation();
+                      console.log("Not yet implemented");
```

Comment:
Just curious what the feature flag status of this is. I'm seeing a Feature.MODIFY_PERMIT_CONDITIONS (separate from Feature.PERMIT_CONDITIONS_PAGE) that should possibly be blocking these buttons.

Reply:
Isn't this whole page behind a feature flag at this point in time?  I should check that.

Reply:
Technically the page is behind PERMIT_CONDITIONS_PAGE (which is currently off) but we have modifications controlled by the MODIFY flag separately. So we could make it possible to _view_ data and not _change_ anything. I think it would be nice to be able to hide the unimplemented features, specifically.